### ISSUE TITLE
RQ: flushing events can impact task processing performance 

### ISSUE THREAD

**Issue by arikfr** [getsentry/sentry-python#570]
RQ worker will wait for the horse to exit before it can execute another task. The following code will block horse's exit to send events:

https://github.com/getsentry/sentry-python/blob/4b4b1c255c953b44f8c2502a7d281b434ab22512/sentry_sdk/integrations/rq.py#L62-L66

This is a non issue when using Sentry to track error only -- events will only be sent if there was an error (hopefully rarely happens). But this becomes an issue when you use Sentry's APM, because it means for every task executed it will pause to send events. 

I don't know how long it takes to deliver events, but in a high volume environment this can become a problem. Though I should bring this to your attention.

**Comment by untitaker** [comment]
You're 100% right. I believe the model needs to be revisited for RQ.

**Comment by untitaker** [comment]
FWIW AWS Lambda has the same problem, conceptually.

**Comment by arikfr** [comment]
> FWIW AWS Lambda has the same problem, conceptually.

It's not entirely the same: AWS Lambda is dynamic and will process the next request with a new process. It costs you more (because Sentry extends the processing time), but not impacting your throughput.

**Comment by untitaker** [comment]
@arikfr how did you end up working around this? Was scaling up worker processes an adequate workaround? Since the rq child proc is constantly blocked on IO I imagine there's some CPU load that ends up being unused.

**Comment by arikfr** [comment]
>  how did you end up working around this? 

Unfortunately we haven't.

**Comment by antonpirker** [comment]
I put this in the internal backlog with low prio. We will decide if and when we are able to tackle this.

Any help is greatly appreciate, so If you know RQ and want to fix this, please go ahead, I am happy to help and review PRs.

**Comment by antonpirker** [comment]
As this has not been reported again or thumbs-upped more, I guess it was not an issue in the last couple of years. 
I will close this issue. If the problem arises in the future we can reopen this issue.
